apps/airdrop/components/claim/ClaimTokenScreen.tsx

```diff
@@ -0,0 +1,90 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { useAccount } from 'wagmi';
+
+import { AlreadyClaimedStep } from './components/AlreadyClaimedStep';
+import { DonationConfirmationStep } from './components/DonationConfirmationStep';
+import type { DonationPercentage } from './components/DonationSelectionStep';
+import { DonationSelectionStep } from './components/DonationSelectionStep';
+import { NotQualifiedStep } from './components/NotQualifiedStep';
+import { ShowClaimableTokensStep } from './components/ShowClaimableTokensStep';
+import { StartClaimStep } from './components/StartClaimStep';
+import { TokenClaimSuccessStep } from './components/TokenClaimSuccessStep';
+
+const DEV_TOKEN_AMOUNT = 2500;
+
+export type ClaimTokenStep =
+  | 'start_claim'
+  | 'show_claimable_tokens'
+  | 'donation_selection'
+  | 'donation_confirmation'
+  | 'token_claim_success'
+  | 'not_qualified'
+  | 'already_claimed';
+
+export function ClaimTokenScreen() {
+  const [step, setStep] = useState<ClaimTokenStep>('start_claim');
+  const [alreadyClaimed, setAlreadyClaimed] = useState(false);
+  const [notQualified, setNotQualified] = useState(false);
+
+  const [donationPercentage, setDonationPercentage] = useState<DonationPercentage>('donate_half');
+
+  const { address = '' } = useAccount();
```

Comment:
we could use the "isConnecting" prop that gets returned to show a loading component